File: microservices/tasks-service/domain/task.go
package domain

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Task struct {
	Id          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Name        string             `bson:"name" json:"name"`
	Description string             `bson:"description" json:"description"`
	Status      Status             `bson:"status" json:"status"`
	ProjectID   string             `bson:"project_id" json:"project_id"`
	Members     []User             `bson:"members" json:"members"`
}

type Tasks []*Task

File: frontend/src/app/model/task.ts
import { Status } from "./status";
import {UserFP} from "./userForProject";

export class Task{
    id : string | null;
    name: string;
    description: string;
    status: string;
    project_id?: string;
  members:UserFP[];

  constructor(
    id : string | null,
    name: string,
    description: string,
    status: string,
    projectId: string,
    members:UserFP[],
  ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.status = status;
        this.project_id = projectId;
        this.members = members;
    }

  //   get statusText(): string {
  //     return Status[this.status];
  // }

}
